internal/actions/cmd/changefinder/main.go

```diff
@@ -0,0 +1,149 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/fs"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+func main() {
+	rootDir, err := os.Getwd()
+	if err != nil {
+		log.Fatal(err)
+	}
+	if len(os.Args) > 1 {
+		rootDir = os.Args[1]
+	}
+	log.Printf("Root dir: %q", rootDir)
+	var modDirs []string
+	// Find all external modules
+	filepath.WalkDir(rootDir, func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+		if d.Name() == "go.mod" && !strings.Contains(path, "/internal/") {
```

Comment:
Can check if `Name` is `internal` and return `SkipDir` if so.

Reply:
Done